pkg/sessions/redis/redis_store.go

```diff
@@ -128,9 +132,20 @@ func (store *SessionStore) Save(rw http.ResponseWriter, req *http.Request, s *se
 		return err
 	}
 
+	se := &envelope.StoreEnvelope{
+		Type:       envelope.RedisType,
```

Comment:
Does knowing the type help?

Reply:
My thought process here was a having this only added a few bytes to the MessagePack encoded message -- with the benefit being on decode operations as customers potentially switching between session store types.

Instead of trying to throw errors decoding the raw bytes in the data field when types are mismatched, use the Type header to know exactly how to decode a session that comes in.

Reply:
This makes sense, SGTM